src/api/define.js

```diff
@@ -122,13 +133,16 @@ export default function (name, opts) {
   Ctor[$props] = createInitProps(Ctor);
   Ctor[$renderer] = createRenderer(Ctor);
 
-  if (customElementsV1) {
-    window.customElements.define(uniqueName, Ctor, { extends: Ctor.extends });
-  } else if (customElementsV0) {
-    return document.registerElement(uniqueName, Ctor);
-  } else {
-    throw new Error('Skate requires native custom element support or a polyfill.');
+  let res;
+  try {
+    res = registerElement(uniqueName, Ctor);
+  } catch(e) {
+    uniqueName = generateUniqueName(name);
```

Comment:
Thanks for explaining IRL :) it's a shame that it sounds like we can't use the idea/function linked above, because it's a bit confusing that the first call of `generateUniqueName(name)` doesn't actually return a unique name.